### ISSUE TITLE
LaTeX (tex) syntax highlighting not active

### ISSUE THREAD

**Issue by wolfv** [jupyterlab/jupyterlab#3858]
I am sure the editor has support for it, but currently `tex` files seem to not get highlighted.

I tried to verify this behaviour in the official binder.

![screenshot from 2018-02-12 13-47-30](https://user-images.githubusercontent.com/885054/36097456-5a45c2e4-0ffb-11e8-9536-4b0fee726562.png)

**Comment by ian-r-rose** [comment]
Yes, for some reason CodeMirror's mode-for-filename logic was missing `.tex` files. You can select the LaTeX mode manually using the View menu.

I submitted a PR for this in CodeMirror, which is now in master, but is unreleased. Until then, maybe we could special-case `.tex` files in core JupyterLab.

**Comment by scimax** [comment]
Wouldn't it be nice to be able to edit the file extension association from the jupyter lab "settings"? Because of your commit, @ian-r-rose , I just found out where I have to add the custom file extension manually. It's just an idea.

**Comment by ian-r-rose** [comment]
This is a nice idea. We can special-case the `jupyterlab_settings` and `jupyterlab_workspaces` files to use the json mode. Can you open a separate issue for this, @scimax?
